site/src/pages/components/switch.explainer.mdx

```diff
@@ -0,0 +1,448 @@
+| `form`        | `form`         | `string`  | `""`          | Associates the element with a form in the document whose `id` is this value.                                                                                                                                                   |
+
+#### Methods
+
+<div class="question">
+  The switch state can be toggled by setting `mySwitch.value = true`. Is a dedicated method
+  `mySwitch.toggle(): (force?: boolean) => void` necessary or helpful?
+</div>
+
+#### Events
+
+| Event Name | Detail Type | Bubbles | Composed | Cancellable | Dispatch Behavior                                      |
+| ---------- | ----------- | ------- | -------- | ----------- | ------------------------------------------------------ |
+| `change`   | none        | `true`  | `true`   | `false`     | Fired when the switch `value` changes.                 |
+| `input`    | none        | `true`  | `true`   | `false`     | Fired when the switch `value` is commited by the user. |
+
+### Appearance
+
+_Screenshots and/or description of the basic appearance of the component._
+
+### Anatomy
+
+{/* _Outline the component's structure with a diagram of its render tree. Enumerate key areas of visual composition and customization._ */}
+
+#### Switch with moving track
+
+<div class="question">
+  If the track content on either side can be of arbitrary length, the track width needs to be `2 *
+  longest side` and the switch needs to be `track width / 2` (not accounting for thumb width,
+  paddings and borders). Is it possible to style this?
+</div>
+
+![Switch anatomy with moving track](../images/switch-moving-track-anatomy.png)
+_Source: [Figma source of switch anatomy with moving track](https://www.figma.com/file/uAYj6yVcAGD2OgjwCTqRol/Switch?type=design&node-id=2%3A28&mode=design&t=OAr7EJBXAQ4AEwfI-1)_
+
+Limitations:
+
+- The thumb needs to be at least the height of the track in order to hide the boundary between active and inactive track side underneath.
+- It's unclear how to style the track and slider if the track content can be of arbitrary length.
+
+#### Switch with static track
+
+<div class="question">
+  Should the thumb width be dinamically adapted to the largest track content width, if any? This
+  woudl ensure that the thumb is always able to hide active and inactive track content.
+</div>
+
+![Switch anatomy with static track](../images/switch-static-track-anatomy.png)
+_Source: [Figma source of switch anatomy with static track](https://www.figma.com/file/uAYj6yVcAGD2OgjwCTqRol/Switch?type=design&node-id=2%3A28&mode=design&t=OAr7EJBXAQ4AEwfI-1)_
+
+Limitations:
+
+- The content on both sides cannot be wider than the tumb. Otherwise, the thumb is no longer capable of hiding the content of the inactive track content.
```

Comment:
@scottaohara IMO, this would go like so:

`<switch>On</switch>`

That said @gfellerph this highlights that you may actually want two parts for on/off. The browser will then handle styling when they are toggled. So something like this:

```
<switch>
      # --- UA Shadow <label>
            <toggled>On</toggledlabe>
            <untoggled>Off</untoggled>
      </label>
</switch>
```

I hit this same issue when sketching out `<checkbox>` as you want not only localization but also people will want to put SVGs and other content within the different states declaratively.

Reply:
Added this to the anatomy section, thanks for the input.